app/lib/vendor_api_specification.rb

```diff
@@ -15,8 +18,33 @@ def as_hash
   end
 
   def spec
-    YAML
-      .load_file("config/vendor_api/v#{@version}.yml")
-      .deep_merge(YAML.load_file("config/vendor_api/experimental-v#{@version}.yml"))
+    (0..minor_version_number(@version)).inject({}) do |versions, minor_version|
```

Comment:
this method is quite long, could we break out into smaller methods to make it easier to parse and follow?